pkg/pipeline/templates/controller.go.tmpl

```diff
@@ -102,6 +103,7 @@ func Setup(mgr ctrl.Manager, o tjcontroller.Options) error {
 		managed.WithInitializers(initializers),
 		managed.WithConnectionPublishers(cps...),
 		managed.WithPollInterval(o.PollInterval),
+		managed.WithMetricRecorder(o.MetricOptions.MRMetrics),
```

Comment:
Let's make this conditional on the availability of the metrics configurations. This way, the change is not backwards-compatible and we will be forcing the provider authors to configure the metric publishing at another site (in this case the provider's main module). So, we introduce a coupling between these two code sites.